### PR TITLE
Feat/bud 262/split by items

### PR DESCRIPTION
**Pull request by Torious**
### BUD Link
[_Link to the bud on Jira_](https://jira.budgetlens.tech/browse/BUD-262)

### Include api paths (if applicable)
api/splititem/ (used to post a list of item split objects)
items/receipt/<int:receipt_id>/ (used to get a list of item objects with receipt_id)

### Summary of the PR
Created Get api to get the list of items in a specific receipt and the post request for when the user splits these items with other users.

### Checks
- [x] Tested Changes
- [x] Ensured that changes do not affect other applications and its use (if applicable)
- [x] Swagger depicts your api accurately (if applicable)
- [x] Tests are created and working (if applicable)

### CODE REVIEW

File: item_split/tests.py [hunk 1]
```diff
@@ -75,223 +72,51 @@ def setUp(self):
...
+            data={'item_list': [{
+                'item_id': self.item2.pk,
                 'shared_user_ids': f'{self.user2.pk}, {self.user3.pk}',
-                'is_shared_with_item_user': False
+                'is_shared_with_item_user': True}]
             },
             format='json'
         )
-
+        print(response)
```
[the remark this answers is not in the document]
Comment (Torious) [reply item_split/tests.py]: done.

File: item_split/tests.py [hunk 2]
```diff
@@ -75,223 +72,51 @@ def setUp(self):
...
+            price=10.15
+        )
+
         # the urls
         self.url_add_item_split = reverse('add_item_split')
 
         # Authenticate user before each test
         self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token.key)
 
-    def test_add_item_split_invalid_users(self):
```
[the remark this answers is not in the document]
Comment (Torious) [reply item_split/tests.py:84]: When i refactored we stopped using specific views so the tests were removed as well.

[the remark this answers is not in the document]
Comment (mamineahmadi) [reply item_split/tests.py:84]: but these URLs still exist: add_item_split and get_user_list so like that means the views for them should still exist and if they still do, why would the tests be removed?

[the remark this answers is not in the document]
Comment (Torious) [reply item_split/tests.py:84]: get_user_list is no longer used, the add_item_split is the one being tests

[the remark this answers is not in the document]
Comment (mamineahmadi) [reply item_split/tests.py:84]: check the urls.py, it still exists there and some tests that you removed used it.

**Review by mamineahmadi — APPROVED** [review]

**Review by massimolopez11 — APPROVED** [review]

**Review by ThatJustin — APPROVED** [review]
All good, nice job.

**Pull request merged**